src/controls/polycontrol.class.ts

```diff
@@ -0,0 +1,149 @@
+import { Point } from '../point.class';
+import { Control } from './control.class';
+import { TMat2D } from '../typedefs';
+import { Polyline } from '../shapes/polyline.class';
+import { multiplyTransformMatrices, transformPoint } from '../util/misc/matrix';
+import { TPointerEvent, Transform } from '../EventTypeDefs';
+import { normalizePoint } from './util';
+import { degreesToRadians } from '../util/misc/radiansDegreesConversion';
+import { TransformActionHandler } from '../EventTypeDefs';
+
+export class PolyControl extends Control {
+  pointIndex: number;
+
+  constructor(options: Partial<Control>, pointIndex: number) {
+    super(options);
+    this.pointIndex = pointIndex;
+  }
+
+  static applySkew(point: Point, shear: Point) {
+    const skewedPoint = new Point();
+    skewedPoint.y = point.y + point.x * shear.y;
+    skewedPoint.x = point.x + skewedPoint.y * shear.x;
+    return skewedPoint;
+  }
+
+  static removeSkew(point: Point, shear: Point) {
+    const unskewedPoint = new Point();
+    unskewedPoint.x = point.x - point.y * shear.x;
+    unskewedPoint.y = point.y - unskewedPoint.x * shear.y;
+    return unskewedPoint;
+  }
+
+  // This function locate the controls.
+  // It'll be used both for drawing and for interaction
+  positionHandler(dim: Point, finalMatrix: TMat2D, polyObject: Polyline) {
+    const x = polyObject.points[this.pointIndex].x - polyObject.pathOffset.x,
+      y = polyObject.points[this.pointIndex].y - polyObject.pathOffset.y;
+    return transformPoint(
+      { x: x, y: y },
+      multiplyTransformMatrices(
+        polyObject.canvas?.viewportTransform ?? ([1, 0, 0, 1, 0, 0] as TMat2D),
+        polyObject.calcTransformMatrix()
+      )
+    );
+  }
+
+  // This function define what the control does.
+  // It'll be called on every mouse move after a control has been clicked and is being dragged.
+  // The function receive as argument the mouse event, the current trasnform object
+  // and the current position in canvas coordinate
+  // transform.target is a reference to the current object being transformed,
+  static polyActionHandler(
+    eventData: TPointerEvent,
+    transform: Transform,
+    x: number,
+    y: number
+  ) {
+    const poly = transform.target as Polyline,
+      currentControl = poly.controls[poly.__corner] as PolyControl,
+      mouseLocalPosition = normalizePoint(
+        poly,
+        new Point(x, y),
+        'center',
+        'center'
+      ),
+      polygonBaseSize = new Point(poly.width, poly.height),
+      size = poly._getTransformedDimensions({}),
```

Comment:
```suggestion
      size = poly._getTransformedDimensions(),
```

Reply:
Typescript complains because options is not an optional parameter in `polyline#_getTransformedDimensoins()`

https://github.com/fabricjs/fabric.js/blob/720a984e0cb5e08a56574a87e09ce6de30b3b485/src/shapes/polyline.class.ts#L153

Maybe it's better to leave it optional in this PR: https://github.com/fabricjs/fabric.js/pull/8374

Reply:
yes